### DISCUSSION TITLE
Using DryIoc with the HostApplicationBuilder

### DISCUSSION THREAD

**Discussion by kateryna-novak** [dadhi/DryIoc#664] [Q&A]
.NET 6.0 introduced a new class for non-web app host building, `HostApplicationBuilder`. However, I cannot find documentation on configuring DryIoc as a service provider factory. For the old `HostBuilder` (`IHostBuilder`), everything is pretty straightforward: 
`hostBuilder.UseServiceProviderFactory(new DryIocServiceProviderFactory())`

And that's it. However, I cannot find anything similar for the `HostApplicationBuilder`. Am I missing something very easy, or is this new class not supported yet?

Thank you in advance for your help!

**Comment by dadhi** [comment]
@kateryna-novak Hi, thanks for the question. I will try to check it myself and get back to you.

**Reply by dadhi** [reply]
@kateryna-novak I have put together the example Console App. Check it here https://github.com/dadhi/DryIoc/blob/master/samples/ConsoleApp/Program.cs

I am using the latest preview version of DryIoc.MS.DI, and suggest you to do so. Soon to be released.
https://github.com/dadhi/DryIoc/blob/master/samples/ConsoleApp/ConsoleApp.csproj
